src/NServiceBus.NHibernate/SynchronizedStorage/NHibernateAmbientTransactionSynchronizedStorageSession.cs

```diff
@@ -51,7 +53,7 @@ public void Dispose()
 
         public async Task CompleteAsync(CancellationToken cancellationToken = default)
         {
-            await onSaveChangesCallback(this, cancellationToken).ConfigureAwait(false);
+            await onSaveChangesCallback(synchronizedStorageSession, cancellationToken).ConfigureAwait(false);
```

Comment:
Because that API:

```
public void OnSaveChanges(Func<ISynchronizedStorageSession, CancellationToken, Task> callback)
```

was already there and I didn't want to change it